api/src/main/java/org/infinispan/api/async/Caches.java

```diff
@@ -0,0 +1,15 @@
+package org.infinispan.api.async;
+
+import java.util.concurrent.CompletionStage;
+
+import org.infinispan.api.configuration.CacheConfiguration;
+
+/**
+ * @author Tristan Tarrant &lt;tristan@infinispan.org&gt;
+ * @since 13.0
+ **/
+public interface Caches {
```

Comment:
Do you envisage templates being part of this interface?

Also, do you plan to expose admin operations via the new api in a similar manner to the current `CacheContainerAdmin`?

Reply:
admin: that's already partially there with the createCache and removeCache.
As for templates, what if we have a `TemplateConfiguration` which implements `Configuration` and do:
```
infinispan.caches().createCache("mycache", new TemplateConfiguration("mytemplate"));
```

Reply:
Your example has made me think that `infinispan.caches().create(..)` & `infinispan.caches().remove(..)` would be simpler as the `caches` suffix is redundant.

For templates, how about:

```java
infinispan.caches().createTemplate(new TemplateConfiguration("mytemplate"));
infinispan.caches().create("mycache", "mytemplate");
infinispan.caches().deleteTemplate(String templateName);
```

Do we need both `TemplateConfiguration` and `CacheConfiguration`? Or are they fundamentally the same?

Reply:
I was actually thinking that we'd have `CacheConfiguration` with two implementations: `ConcreteConfiguration` (horrid name) and `TemplateConfiguration` which would just represent the name of the template, but using the string is much clearer.

Reply:
Done